cmdstanpy/utils.py

```diff
@@ -410,12 +422,14 @@ def read_rdump_metric(path: str) -> List[int]:
         data = fp.read().replace('\n', '')
         m1 = re.search(r'inv_metric\s*<-\s*structure\(\s*c\(', data)
         if not m1:
-            return None
-        m2 = re.search(r'\.Dim\s*=\s*c\(([^)]+)\)', data, m1.end())
-        if not m2:
-            return None
-        dims = m2.group(1).split(',')
-        return [int(d) for d in dims]
+            return_value = None
+        else:
```

Comment:
else is for readability?  the semanticist in my objects, the pragmatist applauds.

Reply:
No, the code behaves differently. It could probably look nicer.

If we return a value within a `with` block, the file is not closed, which causes problems on Windows.